### DISCUSSION TITLE
Properly disposing WebRTC Whep client example

### DISCUSSION THREAD

**Discussion by AllonisDave** [bluenviron/mediamtx#2521] [Questions] [answered]
### Question

In the WebRTC example using the client in an iframe how would one properly dispose of the WhepClient when done with it. I'm trying to dynamically switch streams and just creating a new WhepClient works to switch streams but does not properly clean up the old connection and the MediaMTX server never sees the old stream stop until the browser page is completly closed.

WhepClient has a start method. Is there a stop method that can be applied?

Using MediaMTX 1.2.0 and the javascript examples found by inspecting the content of the iFrame.

Update:

I think the answer was in the ScheduleRestart method. I  added this method

stop() {
    if (this.pc !== null) {
      this.pc.close();
      this.pc = null;
    }
 }

and call it before switching streams and setting the WHEPClient to null. Seems to work. Advise if there is a cleaner way.

**Comment by dillanspencer** [comment, marked as answer]
That is the only way I have found to dispose of a Whep client as well.

[a comment by an automated account is not shown]
